src/badges/views.py

```diff
@@ -61,9 +62,22 @@ def dispatch(self, *args, **kwargs):
 
 
 @allow_non_public_view
-@staff_member_required(login_url='/')
+@login_required
 def badge_create(request):
+
+    # Using `@staff_member_required(login_url='/accounts/login/')` causes a RedirectCycleErrror.
+    # For example, a student that is already logged in tries to access this page,
+    # They will get redirect to `/accounts/login/?next=/achievements/create/`.
+    # And since they are already logged in, they will be redirected to `/achievements/create/`
+    # Causing again to redirect since they are not a staff.
+
+    # We could use `@staff_member_required(login_url='/')` but this breaks the tests
+    # which requires views to be redirected to /accounts/login
```

Comment:
What happens if we just use `@staff_member_required`?  What are we doing elsewhere when we use this decorator?  This view should act the same as everywhere else that uses the same decorator.  And `@login_required` won't work because then students can access this view.

Reply:
With `@staff_member_required`, students that are already logged in are redirected to `/accounts/login/?next=/achievements/create/`, and since they are already logged in, they get redirected to `/achievements/create/` which is has the `@staff_member_required`, so they would get redirected again to `/accounts/login/?next=/achievements/create/`. And it creates a loop...

Reply:
Won't that be a problem for ALL @staff_member_required views?  Maybe what we need is our own decorator that checks if they are already logged in as a student, then they get 403 Permission Denied, and if they are NOT logged in, then redirect to login page.

Reply:
For now, let's just leave it to redirect to django admin log in page.

Reply:
> Won't that be a problem for ALL @staff_member_required views?

For the case of creating badges, I don't think it is. It's the only case that I saw that redirects to `/`, other @staff_member_required redirects to admin.

> Maybe what we need is our own decorator that checks if they are already logged in as a student, then they get 403 Permission Denied, and if they are NOT logged in, then redirect to login page.

I agree and I think that would be better to make a custom decorator. I do think that @staff_member_required should only be used for things that can only be done via admin.